driver/razerchromacommon.c

```diff
@@ -1119,6 +1119,69 @@ struct razer_report razer_chroma_misc_get_dpi_xy_byte(void)
     return report;
 }
 
+/**
+ * Set DPI stages of the device.
+ *
+ * count is the numer of stages to set.
+ * dpi is an array of size 2 * count containing pairs of dpi x and dpi y
+ * values, one pair for each stage.
+ *
+ * E.g.:
+ *   count = 3
+ *   dpi = [ 800, 800, 1800, 1800, 3200, 3200]
+ *         | stage 1 |  stage 2  |  stage 3  |
+ */
+struct razer_report razer_chroma_misc_set_dpi_stages(unsigned char variable_storage, unsigned char count, const unsigned short *dpi)
+{
+    struct razer_report report = get_razer_report(0x04, 0x06, 0x26);
+    unsigned int offset;
+    unsigned int i;
+
+    report.arguments[0] = variable_storage;
+
+    // I think this is a flag that distiguishes fixed DPI from DPI stages.
+    // It's always 0x01 when disabling DPI stages and 0x02 when enabling DPI
+    // them.
+    report.arguments[1] = count >= 2 ? 0x02 : 0x01;
+
+    report.arguments[2] = count;
+
+    offset = 3;
+    for (i = 0; i < count; i++) {
+        // Stage number
+        report.arguments[offset] = i;
+
+        // DPI X
+        report.arguments[offset + 1] = (dpi[0] >> 8) & 0x00FF;
+        report.arguments[offset + 2] = dpi[0] & 0x00FF;
+
+        // DPI Y
+        report.arguments[offset + 3] = (dpi[1] >> 8) & 0x00FF;
+        report.arguments[offset + 4] = dpi[1] & 0x00FF;
+
+        // Reserved
+        report.arguments[offset + 5] = 0;
+        report.arguments[offset + 6] = 0;
```

Comment:
You could use `++offset` (or `offset++` if you start with `report.arguments[offset++] = i;`)  in each line to avoid manually counting up

Reply:
Done, thanks for the suggestion